ball_detection.py

```diff
@@ -0,0 +1,58 @@
+import cv2
+import numpy as np
+import imutils
+
+RADIUS_THRESH = 50
+
+
+class BallDetection:
+
+    def __init__(self, frame):
+        self.frame = frame
+        self.x_val = int(frame.shape[1])
+        self.y_val = int(frame.shape[0])
+
+    def detect_ball(self):
+        frame = cv2.cvtColor(self.frame, cv2.COLOR_BGR2GRAY)
+        self.prep_frame()
+        cnts = cv2.findContours(frame, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_NONE)
+        cnts = imutils.grab_contours(cnts)
+
+        if len(cnts) > 0:
+            for cnt in cnts:
+                ((x, y), radius) = cv2.minEnclosingCircle(cnt)
+                if radius >= RADIUS_THRESH:
+                    # determine the num of border points on the contour (ex: differentiates between circle and rect)
+                    perimeter = cv2.arcLength(cnt, closed=True)
+                    borders = cv2.approxPolyDP(curve=cnt, epsilon=0.02 * perimeter, closed=True)
+                    if len(borders) > 6:
+                        angle = self.find_angle(x, y)
+                        cv2.circle(self.frame, (int(x), int(y)), int(radius), (0, 255, 0), 2)
+
+        return self.frame
+
+    def transform(self, x, y):
+        # transform coordinates to a system with the origin at the bottom-middle of screen
+        dims = self.frame.shape
+        x = x - (dims[1] / 2)
+        y = dims[0] - y
+        return x, y
+
+    def find_angle(self, x, y):
```

Comment:
This is finding the angle with respect to the image right? But in reality we want the angle with respect to the robot. Might be helpful to draw a diagram from the side of the robot to visualize what these look like.

Reply:
We can talk more about this over the next week, doesn't have to be fixed in this PR.